src/validation.cairo

```diff
@@ -38,11 +39,98 @@ impl BlockValidatorImpl of BlockValidator {
 }
 
 #[generate_trait]
-impl TransactionValidatorImpl of TransactionValidator {
+pub impl TransactionValidatorImpl of TransactionValidator {
+    // marker, flag, and witness fields in segwit transactions are not included
+    // this means txid computation is the same for legacy and segwit tx
     fn txid(self: @Transaction) -> u256 {
-        // TODO: implement
-        0
+        let version: felt252 = (*self.version).into();
+        let inputs_count: felt252 = (*self.inputs).len().into();
+        let inputs: Span<TxIn> = *self.inputs;
+        let outputs_count: felt252 = (*self.outputs).len().into();
+        let outputs: Span<TxOut> = *self.outputs;
+        let locktime: felt252 = (*self.lock_time).into();
+
+        // append version (1 byte)
+        let mut sha256_input: ByteArray = "";
+        sha256_input.append_word(version, 1);
```

Comment:
Version is u32, it is already padded.

Reply:
If I am not mistaken.

Reply:
mmh, here is a how version 1 is formatted in tx : 

```
01000000
```

Reply:
just adding version would result in 

```
00000001
```

Reply:
Use `append_word_rev`, or hardcode the values.

Reply:
yes , i use `append_word_rev` now.  Thanks!